File: flutter_theoplayer_sdk_android/android/src/main/kotlin/com/theoplayer/theoplayer/THEOplayerViewNativeFactory.kt
package com.theoplayer.theoplayer

import android.content.Context
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

class THEOplayerViewNativeFactory(private val messenger: BinaryMessenger) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {

    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        val creationParams = args as? Map<String, Any>
        return THEOplayerViewNative(context, viewId, creationParams, messenger)
    }
}

File: flutter_theoplayer_sdk/flutter_theoplayer_sdk_ios/ios/Classes/THEOplayerViewNativeFactory.swift
//
//  THEOplayerViewNativeFactory.swift
//  Runner
//
//  Created by Daniel on 07/04/2023.
//

import Foundation
import Flutter

class THEOplayerNativeViewFactory: NSObject, FlutterPlatformViewFactory {
    private var messenger: FlutterBinaryMessenger

    init(messenger: FlutterBinaryMessenger) {
        self.messenger = messenger
        super.init()
    }

    func create(withFrame frame: CGRect, viewIdentifier viewId: Int64, arguments args: Any?) -> FlutterPlatformView {
        return THEOplayerViewNative(frame: frame, viewIdentifier: viewId, arguments: args, binaryMessenger: messenger)
    }
    
    public func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
        return FlutterStandardMessageCodec.sharedInstance()
    }
    
}
